src/iop/profile_gamma.c

```diff
@@ -304,33 +780,149 @@ void cleanup_global(dt_iop_module_so_t *module)
 {
   dt_iop_profilegamma_global_data_t *gd = (dt_iop_profilegamma_global_data_t *)module->data;
   dt_opencl_free_kernel(gd->kernel_profilegamma);
+  dt_opencl_free_kernel(gd->kernel_profilegamma_log);
   free(module->data);
   module->data = NULL;
 }
+
+
 void gui_init(dt_iop_module_t *self)
 {
   self->gui_data = malloc(sizeof(dt_iop_profilegamma_gui_data_t));
   dt_iop_profilegamma_gui_data_t *g = (dt_iop_profilegamma_gui_data_t *)self->gui_data;
   dt_iop_profilegamma_params_t *p = (dt_iop_profilegamma_params_t *)self->params;
 
+  self->request_color_pick = DT_REQUEST_COLORPICK_OFF;
+
   self->widget = gtk_box_new(GTK_ORIENTATION_VERTICAL, DT_BAUHAUS_SPACE);
 
-  g->linear = dt_bauhaus_slider_new_with_range(self, 0.0, 1.0, 0.0001, p->linear, 4);
-  g->gamma = dt_bauhaus_slider_new_with_range(self, 0.0, 1.0, 0.0001, p->gamma, 4);
+  // mode choice
+  g->mode = dt_bauhaus_combobox_new(self);
+  dt_bauhaus_widget_set_label(g->mode, NULL, _("mode"));
+  dt_bauhaus_combobox_add(g->mode, _("logarithmic"));
+  dt_bauhaus_combobox_add(g->mode, _("gamma"));
+  gtk_box_pack_start(GTK_BOX(self->widget), GTK_WIDGET(g->mode), TRUE, TRUE, 0);
+  gtk_widget_set_tooltip_text(g->mode, _("tone mapping method"));
+  g_signal_connect(G_OBJECT(g->mode), "value-changed", G_CALLBACK(mode_callback), self);
 
-  dt_bauhaus_widget_set_label(g->linear, NULL, _("linear"));
-  dt_bauhaus_widget_set_label(g->gamma, NULL, _("gamma"));
+  // prepare the modes widgets stack
+  g->mode_stack = gtk_stack_new();
+  gtk_stack_set_homogeneous(GTK_STACK(g->mode_stack), FALSE);
+  gtk_box_pack_start(GTK_BOX(self->widget), g->mode_stack, TRUE, TRUE, 0);
 
-  gtk_box_pack_start(GTK_BOX(self->widget), g->linear, TRUE, TRUE, 0);
-  gtk_box_pack_start(GTK_BOX(self->widget), g->gamma, TRUE, TRUE, 0);
 
+  /**** GAMMA MODE ***/
+  GtkWidget *vbox_gamma = GTK_WIDGET(gtk_box_new(GTK_ORIENTATION_VERTICAL, DT_BAUHAUS_SPACE));
+  // linear slider
+  g->linear = dt_bauhaus_slider_new_with_range(self, 0.0, 1.0, 0.0001, p->linear, 4);
+  dt_bauhaus_widget_set_label(g->linear, NULL, _("linear"));
+  gtk_box_pack_start(GTK_BOX(vbox_gamma), g->linear, TRUE, TRUE, 0);
   gtk_widget_set_tooltip_text(g->linear, _("linear part"));
-  gtk_widget_set_tooltip_text(g->gamma, _("gamma exponential factor"));
-
   g_signal_connect(G_OBJECT(g->linear), "value-changed", G_CALLBACK(linear_callback), self);
+
+  // gamma slider
+  g->gamma = dt_bauhaus_slider_new_with_range(self, 0.0, 1.0, 0.0001, p->gamma, 4);
+  dt_bauhaus_widget_set_label(g->gamma, NULL, _("gamma"));
+  gtk_box_pack_start(GTK_BOX(vbox_gamma), g->gamma, TRUE, TRUE, 0);
+  gtk_widget_set_tooltip_text(g->gamma, _("gamma exponential factor"));
   g_signal_connect(G_OBJECT(g->gamma), "value-changed", G_CALLBACK(gamma_callback), self);
+
+  gtk_widget_show_all(vbox_gamma);
+  gtk_stack_add_named(GTK_STACK(g->mode_stack), vbox_gamma, "gamma");
+
+
+  /**** LOG MODE ****/
```

Comment:
What stuff ? What is mislabelled ? I don't understand what you mean

Reply:
Most of the defaults for these sliders in`"camera & signal properties"`.
You should be able to get most of that from the actual black/white levels.
Else i suspect the name is unrepresentable.

Reply:
It's not black level as in the exposure module, it's the limit between black (signal) and noise. Using a log function, values very close to zero (thus noise) get amplified much more than correct values. We need a threshold to cut out very small values before sending them in the log. This depends on the sensor, ISO value, previous denoising and demosaicing, and exposure compensation (gain).

The linear factor helps to center the dynamic range such that black are at - DR / 2, white are at +DR / 2, and the target middle grey is at 0 EV, thus 50 %. It should be measured or estimated by optimization.

Reply:
Again, I'd leave things as a base log. You shouldn't be having any noise amplification at all, and if you are, the code is busted.

Base logs have been around longer than DSLRs. They work.

Reply:
> Again, I'd leave things as a base log. You shouldn't be having any noise amplification at all, and if you are, the code is busted.
> 
> Base logs have been around longer than DSLRs. They work.

I'm sorry, but in dt, this module can take negative values which are going to output NaN or zero division errors. I trust DSLR manufacturer can normalize the values before the log, taking account of the noise level, but we have to take whatever crap the previous modules in the pixelpipe give us.

> Most of the defaults for these sliders in`"camera & signal properties"`.
> You should be able to get most of that from the actual black/white levels.
> Else i suspect the name is unrepresentable.

You were actually right, looking again at the maths, these parameters are dependent of the dynamic range so I just removed them throm the UI.

Reply:
> You were actually right, looking again at the maths, these parameters are dependent of the dynamic range so I just removed them throm the UI.

... are you still confident that dynamic range won't be auto-calculatable the way i suggested in the first comment in this thread?

Reply:
no, this dynamic range is the output dynamic range. You can see that as a target contrast. It can be optimized to fit the whole [0; 1]  range at the output, considering the input extrema (that's the purpose of the last slider + autofix_callback function), but there is no way you could guess it from hadrware properties (and, again, it depends on what you did n the previous modules).